husky_components/data-navigation/main.js

```diff
@@ -237,6 +306,18 @@ define([
             this.sandbox.on(CLEAR_CACHE.call(this), function() {
                 this.cache.deleteAll();
             }.bind(this));
+
+            this.sandbox.on('husky.search.data-navigation', function(searchTerm) {
+                this.searchTerm = searchTerm;
```

Comment:
Is this really necessary? Looks like a source of bugs to me. Can't we just read this value from the text field in the `getCurrentUrl` field?